### PR TITLE
[Test] Migrate the first test to the new infra

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This migrate the simplest nightly test to the new infra. I will also explore k8s migration with this test

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/scripts/run_release_test.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 )
 @click.option(
     "--ray-wheels",
-    default=None,
+    default="master",
```
Comment (rkooo567) [inline release/ray_release/scripts/run_release_test.py:47]: I think it is the reasonable default behavior. Lmk if you disagree!

File: release/nightly_tests/wait_cluster.py [hunk 2]
```diff
@@ -26,15 +26,6 @@
 next_feedback = start
 max_time = start + args.max_time_s
 
-
-def num_alive_nodes():
```
Comment (rkooo567) [inline release/nightly_tests/wait_cluster.py:30]: Looks like `wait_cluster` and `_wait_cluster` is somehow linked?

File: release/ray_release/scripts/run_release_test.py (hunk 1 above)
Comment (krfricke) [reply release/ray_release/scripts/run_release_test.py:47]: I disagree - I'd like the builds to run on the specified commit per default. This makes sense as the checks will be displayed in GitHub and e.g. trigger a red check mark on the commits (e.g. master). It makes it also easier to identify breaking commits.

It's also good for release managers working on a release branch - when they kick off the release tests for this branch they can be sure they are run on the latest commit wheels of this branch, not of some (potentially stale) wheel.

I'd be open to set "master" as the default for manually triggered runs on the master branch (e.g. pre-fill the field with "master" when kicking off).

File: release/nightly_tests/wait_cluster.py (hunk 2 above)
Comment (krfricke) [reply release/nightly_tests/wait_cluster.py:30]: If you add `wait_for_nodes` in the test run configuration, the infra can wait until the nodes are up and you don't need to do this in the script anymore

Comment (krfricke) [reply release/nightly_tests/wait_cluster.py:30]: But if this is dynamic you may want to keep it like it is

Comment (rkooo567) [reply release/nightly_tests/wait_cluster.py:30]: Oh the weird thing is I didn't change this file, but it was changed for some reasons like this..

File: release/ray_release/scripts/run_release_test.py (hunk 1 above)
Comment (rkooo567) [reply release/ray_release/scripts/run_release_test.py:47]: Isn't it possible to just always set this value if the tests are started by the build though? 

In that case, the only difference is that when you run this CLI in the local machine, you don't need to always specify `--ray-wheels master`. Or am I missing something here?

Comment (rkooo567) [reply release/ray_release/scripts/run_release_test.py:47]: But anyway let me merge this PR without setting the default! We can keep discussion here

**Comment by rkooo567** [comment]
Test failure should be unrelated. cc @scv119 can you approve and merge it?

**Review by jjyao — APPROVED** [review]

**Review by scv119 — APPROVED** [review]

File: release/release_tests.yaml [hunk 3]
```diff
@@ -761,3 +761,26 @@
...
+  cluster:
+    cluster_env: shuffle/shuffle_app_config.yaml
+    cluster_compute: shuffle/shuffle_compute_single.yaml
+
+  run:
+    timeout: 3000
+    script: python shuffle/shuffle_test.py --num-partitions=50 --partition-size=200e6
+    
+    type: sdk_command
+    file_manager: sdk
```
Comment (jjyao) [inline release/release_tests.yaml:786]: Could you also migrate microbenchmark first as well so that I can start to set up alerts for it?

Comment (rkooo567) [reply release/release_tests.yaml:786]: Will make a follow up

**Comment by rkooo567** [comment]
Failed tests should be unrelated

File: release/release_tests.yaml (hunk 3 above)
Comment (jjyao) [reply release/release_tests.yaml:786]: Should you also add frequency in the follow up PR?

**Pull request merged**